packages/mobile/src/services/offline-downloader/offline-sync.ts

```diff
@@ -160,14 +161,29 @@ export const syncCollectionTracks = async (
     })
   )?.[0]
 
-  // TODO: will discovery serve a removed playlist?
   if (!updatedCollection) return
-  const downloadedCollectionTrackIds = offlineCollection.tracks
-    ?.map((track) => track.track_id)
-    ?.filter((trackId) => downloadedTracks[trackId] != null)
+
+  const downloadedCollectionTrackIds =
+    offlineCollection.tracks
+      ?.map((track) => track.track_id)
+      ?.filter((trackId) => downloadedTracks[trackId] != null) ?? []
+
+  if (updatedCollection.is_delete) {
+    removeCollectionDownload(
+      collectionId,
+      downloadedCollectionTrackIds.map((trackId) => ({
```

Comment:
@dylanjeffers @amendelsohn , not sure if it made more sense to use `downloadedCollectionTrackIds` here or `downloadedOrQueuedCollectionTrackIds`. (This is for the `tracksForDownload` param for `removeCollectionDownload`.)

Reply:
good q, def defer to @amendelsohn on this one

Reply:
Either one is likely fine, but we have a check in `downloadCollection` for is_delete against the latest version from discovery, so I think it's fine to just do the downloaded ones and not the queued ones. `removeCollectionDownload` does cancel the queued job, so it should work fine either way. The advantage of doing it now is you don't have to wait for the queue to drain, but it will cost more up front.

Overall, my instinct is to use `downloadedCollectionTrackIds`, but I do think either would be functional.